spec.html

```diff
@@ -15692,21 +15790,79 @@ <h1>Runtime Semantics: BindingInitialization</h1>
+
+        <emu-note>These collect a list of all bound property names rather than just empty completion.</emu-note>
+
         <emu-grammar>BindingPropertyList : BindingPropertyList `,` BindingProperty</emu-grammar>
         <emu-alg>
-          1. Perform ? BindingInitialization for |BindingPropertyList| using _value_ and _environment_ as arguments.
-          1. Return the result of performing BindingInitialization for |BindingProperty| using _value_ and _environment_ as arguments.
+          1. Let _boundNames_ be the result of performing PropertyBindingInitialization for |BindingPropertyList| using _value_ and _environment_ as arguments.
+          1. ReturnIfAbrupt(_boundNames_).
+          1. Let _nextNames_ be the result of performing PropertyBindingInitialization for |BindingProperty| using _value_ and _environment_ as arguments.
+          1. ReturnIfAbrupt(_nextNames_).
+          1. Append each item in _nextNames_ to the end of _boundNames_.
+          1. Return _boundNames_.
         </emu-alg>
+
         <emu-grammar>BindingProperty : SingleNameBinding</emu-grammar>
         <emu-alg>
           1. Let _name_ be the string that is the only element of BoundNames of |SingleNameBinding|.
-          1. Return the result of performing KeyedBindingInitialization for |SingleNameBinding| using _value_, _environment_, and _name_ as the arguments.
+          1. Let _status_ be the result of performing KeyedBindingInitialization for |SingleNameBinding| using _value_, _environment_, and _name_ as the arguments.
+          1. ReturnIfAbrupt(_status_).
+          1. Return a new List containing _name_.
         </emu-alg>
+
         <emu-grammar>BindingProperty : PropertyName `:` BindingElement</emu-grammar>
         <emu-alg>
           1. Let _P_ be the result of evaluating |PropertyName|.
           1. ReturnIfAbrupt(_P_).
-          1. Return the result of performing KeyedBindingInitialization for |BindingElement| using _value_, _environment_, and _P_ as arguments.
+          1. Let _status_ be the result of performing KeyedBindingInitialization of |BindingElement| with _value_, _environment_, and _P_ as the arguments.
+          1. ReturnIfAbrupt(_status_).
+          1. Return a new List containing _P_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="Rest-RuntimeSemantics-RestBindingInitialization">
+        <h1>Runtime Semantics: RestBindingInitialization</h1>
+        <p>With parameters _value_, _environment_ and _excludedNames_.</p>
+
+        <emu-grammar>BindingRestProperty : `...` BindingIdentifier</emu-grammar>
+        <emu-alg>
+          1. Let _restObj_ be ObjectCreate(%ObjectPrototype%).
+          1. Let _assignStatus_ be CopyDataProperties(_restObj_, _value_, _excludedNames_).
+          1. ReturnIfAbrupt(_assignStatus_).
+          1. Let _bindingId_ be StringValue of |BindingIdentifier|.
+          1. Let _lhs_ be ResolveBinding(_bindingId_, _environment_).
+          1. ReturnIfAbrupt(_lhs_).
```

Comment:
This line should be removed, and the preceding `ResolveBinding ` be prefixed with a `?`